frontend/packages/dev-console/src/components/import/__tests__/import-submit-utils.spec.ts

```diff
@@ -486,4 +488,147 @@ describe('Import Submit Utils', () => {
+      expect(telGHData).toEqual({ ...ghImportTelData, useAdvancedOptionsScaling: true });
+
+      const telGhScalingData = submitUtils.isScalingAdvOptions(
+        ghImportAdvData.resources,
+        ghImportAdvData.deployment,
+        ghImportAdvData.serverless,
+      );
+
+      expect(telGhScalingData).toEqual(true);
+    });
+
+    it('getTelemetryImport should return appropriate data with useAdvancedOptionsScaling option as true if scaling advanced options are used for Serverless', () => {
+      const ghImportAdvData = {
+        ...ghImportDefaultData,
+        resources: Resources.KnativeService,
+        serverless: {
+          ...ghImportDefaultData.serverless,
+          scaling: {
+            ...ghImportDefaultData.serverless.scaling,
+            concurrencytarget: 102,
+          },
+        },
+      };
+      const telGHData = submitUtils.getTelemetryImport(ghImportAdvData);
+      expect(telGHData).toEqual({
+        ...ghImportTelData,
+        resource: 'Knative Service',
+        useAdvancedOptionsScaling: true,
+      });
+
+      const telGhScalingData = submitUtils.isScalingAdvOptions(
+        ghImportAdvData.resources,
+        ghImportAdvData.deployment,
+        ghImportAdvData.serverless,
+      );
+
+      expect(telGhScalingData).toEqual(true);
+    });
+
+    it('getTelemetryImport should return appropriate data with useAdvancedOptionsRoute option as true if route advanced options are used', () => {
+      const ghImportAdvData = {
+        ...ghImportDefaultData,
+        route: {
+          ...ghImportDefaultData.route,
+          hostname: 'domain.org',
+        },
+      };
+      const telGHData = submitUtils.getTelemetryImport(ghImportAdvData);
+      expect(telGHData).toEqual({ ...ghImportTelData, useAdvancedOptionsRoute: true });
+
+      const telGhScalingData = submitUtils.isRouteAdvOptionsUsed(
+        ghImportAdvData.resources,
+        ghImportAdvData.route,
+        ghImportAdvData.serverless,
+      );
+
+      expect(telGhScalingData).toEqual(true);
+    });
+
+    it('getTelemetryImport should return appropriate data with useAdvancedOptionsRoute option as true if route advanced options are used for Serverless', () => {
+      const ghImportAdvData = {
+        ...ghImportDefaultData,
+        resources: Resources.KnativeService,
+        serverless: {
+          ...ghImportDefaultData.serverless,
+          domainMapping: ['domain.org'],
+        },
+      };
+      const telGHData = submitUtils.getTelemetryImport(ghImportAdvData);
+      expect(telGHData).toEqual({
+        ...ghImportTelData,
+        resource: 'Knative Service',
+        useAdvancedOptionsRoute: true,
+      });
+
+      const telGhScalingData = submitUtils.isRouteAdvOptionsUsed(
+        ghImportAdvData.resources,
+        ghImportAdvData.route,
+        ghImportAdvData.serverless,
+      );
+
+      expect(telGhScalingData).toEqual(true);
+    });
+  });
```

Comment:
@invincibleJai  can you also add a test for `devFileLanguage` and `devFileProjectType` props as well?

Reply:
have added specs for devfile, PTAL